core/math/math_funcs.h

```diff
@@ -447,14 +447,14 @@ class Math {
 
 	static _ALWAYS_INLINE_ double smoothstep(double p_from, double p_to, double p_s) {
 		if (is_equal_approx(p_from, p_to)) {
-			return p_from;
+			return p_s <= p_from ? 0.0 : 1.0;
```

Comment:
Almost equal `p_from > p_to` is still technically a "negative range" hence I'd say it should be treated as such. So e.g. `smoothstep(0.000001, 0.0, 0.0)` should be `1.0`, not `0.0`.
```suggestion
			return p_s <= MIN(p_from, p_to) ? 0.0 : 1.0;
```

Reply:
Your example will still return 0.0, even with the modified code as `0.0 <= 0.0` will evaluate to `true`

Reply:
I was figuring ranges smaller than is_equals_approx could be treated as a zero-range; however this is a good improvement. If the range is smaller than the is_equals_approx window then the S-curve segment disappears and it turns into a binary comparison. This makes that binary comparison polarity accurate down to the level of the float/double comparison.